app/models/project_step.rb

```diff
@@ -384,16 +386,20 @@ def handle_finalized_at
     end
   end
 
+  # Copies date changes to schedule_children as appropriate.
   def handle_schedule_children
     return unless persisted? && schedule_children.present? &&
-      (scheduled_start_date_changed? || scheduled_duration_days_changed?)
+      (scheduled_start_date_changed? || scheduled_duration_days_changed? || actual_end_date_changed?)
 
     schedule_children.each do |step|
-      step.scheduled_start_date = scheduled_end_date
-      step.save
+      step.scheduled_start_date = display_end_date + 1
```

Comment:
It seems `display_end_date` could be nil here as well, if for example the start date and duration on an incomplete step were both set to blank. 

I'm also wondering if it might be better to do something like `schedule_children.each(&: copy_schedule_parent_date)` instead, so as to keep that logic in only one place. Slower though, I suppose.